### PR TITLE
Feat: Added filters to selector

### PR DESCRIPTION
**Pull request by alex-a-ncino**
#### Change Summary:

- Added support for filters
- Removed filterByIdentities

#### Areas of Direct/Downstream Impact:

-

#### Other Notes:

-

#### Merge Checklist (Required):

- [ ] As the Developer, I certify that I have thoroughly tested and self-reviewed my code.
- [x] As the Reviewer, I certify that I have reviewed this code and believe it is (virtually) free of issues and ready to be merged.
- [ ] The package.json version number has been incremented in this pull request.

### CODE REVIEW

File: src/extensions/args-extension.ts [hunk 1]
```diff
@@ -35,5 +35,13 @@ module.exports = (toolbox: GluegunToolbox) => {
     return parameters.options.api || parameters.options.a || '47.0';
   }
 
-  toolbox.args = { parseOptions, getOutputDirectory, getApiVersion };
+  function getSelectorFilters(): any {
```
Comment (callietoothman) [inline src/extensions/args-extension.ts]: I don't think this should be in the extension. It's specific to the Selector command and can live there.

File: src/commands/new/selector.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
-import { GluegunToolbox } from 'gluegun';
+import { GluegunToolbox, GluegunCommand } from 'gluegun';
 import { OptionAttribute } from '../../extensions/args-extension';
+import * as meta from '../new/meta';
 
 const SELECTOR_OPTIONS: OptionAttribute[] = [
   { name: 'name', alias: 'n', required: true },
```
Comment (callietoothman) [inline src/commands/new/selector.ts]: Should require add `filters` as optional command.

File: src/templates/selector.cls.ejs [hunk 3]
```diff
@@ -1,22 +1,16 @@
 public with sharing class <%= props.componentName %>Selector extends SObjectsSelector {
 
-	public <%= props.componentName %>Selector filterByIdentities(nFORCE.Query.Builder query, String[] identities) {
-		if (identities == null || identities.isEmpty()) {
-			return this;
+	<% props.filters.forEach(function(filter) { %>
```
Comment (callietoothman) [inline src/templates/selector.cls.ejs]: Add an underscore after to `<%_` tag to remove the extra tab from the generated file.
```suggestion
	<%_ props.filters.forEach(function(filter) { %>
```

File: src/templates/selector.cls.ejs [hunk 4]
```diff
@@ -1,22 +1,16 @@
...
+	public nFORCE.Query.Builder filterBy<%= filter.name %>(nFORCE.Query.Builder query, <%= filter.type %> <%= filter.param %>) {
```
Comment (callietoothman) [inline src/templates/selector.cls.ejs]: I don't think we need separate attributes for `filter.name` and `filter.param`. We can derive the method name from the param. For instance, if the param was **loanId**, then we could just capitalize it in order to append it to the method name (**filterByLoanId**).

File: src/commands/new/selector.ts [hunk 5]
```diff
@@ -20,20 +21,27 @@ module.exports = {
     try {
       const options = args.parseOptions(SELECTOR_OPTIONS);
       const outputDirectory = args.getOutputDirectory();
+      const filters = args.getSelectorFilters();
```
Comment (callietoothman) [inline src/commands/new/selector.ts]: You may not even need a method for this:
```suggestion
      const filters = JSON.parse(parameters.options.filters) || [];
```

File: src/templates/selector.cls.ejs (hunk 4 above)
Comment (alex-a-ncino) [reply src/templates/selector.cls.ejs]: I thought about that but then I saw scenarios where the parameter is depositIdentifiers but the method is filterByDeposits

Comment (alex-a-ncino) [reply src/templates/selector.cls.ejs]: `	public nFORCE.Query.Builder filterByDeposits(nFORCE.Query.Builder query,
		List<String> depositIdentifiers) {`

File: src/commands/new/selector.ts [hunk 6]
```diff
@@ -20,20 +22,27 @@ module.exports = {
     try {
       const options = args.parseOptions(SELECTOR_OPTIONS);
       const outputDirectory = args.getOutputDirectory();
+      const filters = parameters.options.filters ? JSON.parse(parameters.options.filters) : [];
```
Comment (callietoothman) [inline src/commands/new/selector.ts]: You don't even need the ternary operator: `JSON.parse(options.filters) || []` will do the same thing.

File: src/templates/selector.cls.ejs [hunk 7]
```diff
@@ -1,22 +1,14 @@
...
-
-		this.filterBy(
-			query,
-			nFORCE.Query.condition(<%= props.sObjectType %>.Id, ids.size() == 1 ? '=' : 'IN', ids)
-		);
-		return this;
+		return query;
 	}
-
+	<% }); %>
```
Comment (callietoothman) [inline src/templates/selector.cls.ejs]: This will get rid of the extra tab and newline at the bottom of each method.
```suggestion
	<%_ }); -%>
```

**Pull request merged**